pkg/provision/engine.go

```diff
@@ -81,6 +87,23 @@ func (e *Engine) Run(ctx context.Context) error {
 
 	cReservation := e.source.Reservations(ctx)
 
+	// Indicates wether provisiond cleaned up resources after
+	// booting and provisioning all reservations
+	triggeredCleanupOnBoot := false
```

Comment:
Why didn't you name it `triggerCleanup = true` instead! this is a very weird naming that makes it confusing. and then you have to do the negation in the `if` check. Instead it can be `if reservation.last && triggerCleanup` then u change the value to false.

Reply:
It's the past tense of `trigger`. It means if provisiond HAS triggered cleanup before. We only want to trigger cleanup at boot.